.github/workflows/release.yaml

```diff
@@ -0,0 +1,34 @@
+name: Release
+
+on:
+  create:
```

Comment:
I think create makes more sense, since you may don't want to release a new binary each time you push on the tagged branch. You might want to decide when you release a new package.